src/pb_stub.cc

```diff
@@ -1092,9 +1083,14 @@ Stub::ParentToStubMQMonitor()
     }
 
     {
-      std::lock_guard<std::mutex> lock(response_generator_map_mu_);
-      response_generator_map_[infer_response->Id()]->EnqueueResponse(
-          std::move(infer_response));
+      std::lock_guard<std::mutex> lock(response_iterator_map_mu_);
+      // No need to enqueue the response if the response iterator has been
+      // cleaned up and removed from the map.
+      if (response_iterator_map_.find(infer_response->Id()) !=
```

Comment:
I think maybe we don't have to wait for retrieving the remaining responses in ResponseIterator destructor if we don't enqueue the responses after clean-up.

Reply:
On second thought, I think I should revert this change as it's possible that there is a new response with the same id just got stored inside the map.